plugin/rewrite/rewrite.go

```diff
@@ -38,6 +38,9 @@ type Rewrite struct {
 
 // ServeDNS implements the plugin.Handler interface.
 func (rw Rewrite) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
+
+	// ensure r is a copy in order to not pollute the initial request
+	r = r.Copy()
```

Comment:
We should not modify the initial query.
But creating  a Copy() for each query may be heavy (CPU / Memory).

Should we Copy ONLY when there will be a change on OPT ?